[13.0][ADD]sale_advance_payment: Backport from 14.0

GuillemCForgeFlow opened the pull request:
@ForgeFlow

omar7r committed:
[ADD] sale_advance_payment: Register payments in advance

MateuGForgeFlow committed:
[MIG] sale_advance_payment: Migration to 14.0

JordiBForgeFlow committed:
add field 'advance_payment_status' and improve the payment wizard for better support in multi-currency.

MateuGForgeFlow committed:
Readme Usage and Tests

MateuGForgeFlow committed:
Currency fix and float_compare

oca-travis committed:
[UPD] Update sale_advance_payment.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

remi-filament committed:
Added translation using Weblate (French)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 92.1% (35 of 38 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_advance_payment
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_advance_payment/fr/

GuillemCForgeFlow committed:
[13.0][ADD]sale_advance_payment: Backport from 14.0

JordiBForgeFlow commented:
@JasminSForgeFlow can you help @GuillemCForgeFlow to figure out why travis fails here?

JordiBForgeFlow commented:
The error in Travis seems to be unrelated to this module, but then I can see other pull requests in this repo targeting 13.0 that are green.

JordiBForgeFlow committed:
[imp][sale_advance_payment] various improvements
- Allow to make advance payments when invoice has been created.
  This is justified by the fact that users may want to create a
  payment reversal in some circumstances.
- Introduce payment type (inbound, outbound) to improve usability

AaronHForgeFlow commented on sale_advance_payment/tests/test_sale_advance_payment.py:
```diff
@@ -0,0 +1,247 @@
+
+        # Sale Order
+        cls.sale_order_1 = cls.env["sale.order"].create(
+            {"partner_id": cls.res_partner_1.id}
```
```suggestion
            {"name": "/", partner_id": cls.res_partner_1.id}
```

I think CI fails because the module sale_order_revision works if you create a sales order as a user, but it does not if you create a sales order using create method. If the name is "/" it will generate a new number for the sales order, so the constraint will not raise.

Can you try that?

JordiMForgeFlow replied:
It still fails...it is weird because it is only failing in the OCB tests :/

AaronHForgeFlow replied:
The same error?

JordiBForgeFlow commented:
/ocabot merge nobump

JordiBForgeFlow commented:
/ocabot merge nobump

JordiMForgeFlow committed:
[FIX][sale_advance_payment]: fix test

JordiBForgeFlow commented:
/ocabot merge nobump
